### ISSUE TITLE
[BUG] expect "compound column to be unique" is not scalable

### ISSUE THREAD

**Issue by itaise** [fivetran/great_expectations#6468]
Hi, 
We are using GE over Trino, and one of the main tests we use is ["expect compound column to be unique"](https://greatexpectations.io/expectations/expect_compound_columns_to_be_unique?filterType=Backend%20support&gotoPage=1&showFilters=false&viewType=Summary)

However, we found that this expectation (in its sql alchemy implementation) is not scalable enough for us. 
We are running this expectation on large tables, and it fails on insufficient resources - it seems that [this metric](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/expectations/metrics/multicolumn_map_metrics/compound_columns_unique.py#L117) is creating a join between two huge tables. 

Instead, we would like to have something much lighter like: 
```
select count(1) from (
select a,b,c, count(1)
from table
group by a,b,c
having count(1) > 1
)
```

This way it will work without a join, and be able to calculate on huge dwh tables like ours..
I wonder if we should create a new custom ex., or should this one be altered? 

Thanks a lot!

**Comment by austiezr** [comment]
Hey @itaise ! Thanks for raising this. 

We'd love to see a Custom Expectation with this approach; from there, we can test and review to establish whether this core expectation should be replaced / altered, or the Custom Expectation maintained alongside.

**Comment by itaise** [comment]
Hi @austiezr,  I created an expectation that solves the issue for us, and works well with our large tables on Trino, can you please check? thanks a lot!

**Comment by talagluck** [comment]
Resolved by #6473 - thanks @itaise!
